akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -554,30 +554,35 @@ object ShardCoordinator {
 
     timers.startSingleTimer("hand-off-timeout", ReceiveTimeout, handOffTimeout)
 
-    def receive = {
+    def receive: Receive = {
       case BeginHandOffAck(`shard`) =>
         log.debug("BeginHandOffAck for shard [{}] received from [{}].", shard, sender())
         acked(sender())
-      case ShardRegionTerminated(shardRegion) =>
+      case RebalanceWorker.ShardRegionTerminated(shardRegion) =>
         log.debug("ShardRegion [{}] terminated while waiting for BeginHandOffAck for shard [{}].", shardRegion, shard)
         acked(shardRegion)
       case ReceiveTimeout =>
-        log.debug("Rebalance of [{}]  from [{}] timed out", shard, from)
+        log.debug("Rebalance of shard [{}] from [{}] timed out", shard, shardRegionFrom)
         done(ok = false)
     }
 
     private def acked(shardRegion: ActorRef) = {
       remaining -= shardRegion
       if (remaining.isEmpty) {
         log.debug("All shard regions acked, handing off shard [{}].", shard)
-        from ! HandOff(shard)
+        shardRegionFrom ! HandOff(shard)
         context.become(stoppingShard, discardOld = true)
+      } else {
+        log.debug("Remaining shard regions: {}", remaining)
```

Comment:
`remaining.size` so that the logging doesn't become too verbose

Reply:
👍